cocotb/scheduler.py

```diff
@@ -775,9 +775,13 @@ def finish_scheduler(self, exc):
         """Directly call into the regression manager and end test
            once we return the sim will close us so no cleanup is needed.
         """
-        self.log.debug("Issue sim closedown result to regression object")
-        self._test.abort(exc)
-        cocotb.regression_manager.handle_result(self._test)
+        # If there is an error during cocotb initialization, self._test may not
+        # have been set yet. Don't cause another Python exception here.
```

Comment:
Might be worth adding a `self.log.warn` if this gets reached without `self._test` being set

Reply:
The only place `finish_scheduler()` is called is from `_sim_event()`, which logs an error just before calling.